src/hipscat/catalog/margin_cache/margin_catalog.py

```diff
@@ -46,3 +48,14 @@ def __init__(
         super().__init__(
             catalog_info, pixels, catalog_path=catalog_path, moc=moc, storage_options=storage_options
         )
+
+    def filter_by_moc(self, moc: MOC) -> Self:
```

Comment:
Can you also add a docstring to this method, mypy didn't catch it.

Reply:
it didn't catch it because it's an overwritten method, but I think it's a good idea to update the docstring to include the margin specific behavior.